bash_it.sh

```diff
@@ -109,19 +108,18 @@ done
 # Custom
 BASH_IT_LOG_PREFIX="core: main: "
 _log_debug "Loading general custom files..."
-CUSTOM="${BASH_IT_CUSTOM:=${BASH_IT}/custom}/*.bash ${BASH_IT_CUSTOM:=${BASH_IT}/custom}/**/*.bash"
-for _bash_it_config_file in $CUSTOM; do
+for _bash_it_config_file in "${BASH_IT_CUSTOM:=${BASH_IT}/custom}"/*.bash "${BASH_IT_CUSTOM}"/**/*.bash; do
```

Comment:
Although it's quite clever, I feel its bad practice to hide a default assignment (and immediate re-use) in the middle of a complex statement like this.

You're going to want to either add a clarifying comment so the next dev to review this code won't miss the defaulting assignment, or you can use the logic from my original suggestion:
```
CUSTOM="${BASH_IT_CUSTOM:=${BASH_IT}/custom}"
for _bash_it_config_file in "${CUSTOM}"/*.bash "${CUSTOM}"/**/*.bash; do
```
I prefer my technique as its more clear and matches custom var/default assignment patterns used throughout the framework (ie in themes)

I'm traveling so don't have time review the newer additions to this PR yet, but will try to have a look within the next week.

Thanks !

Reply:
Yes, this is fair. I'll update my patch.